drools-wb-screens/drools-wb-drl-text-editor/drools-wb-drl-text-editor-backend/src/test/java/org/drools/workbench/screens/drltext/backend/server/DRLTextEditorServiceImplCDITest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.drltext.backend.server;
+
+import java.net.URL;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.drools.workbench.screens.drltext.service.DRLTextEditorService;
+import org.guvnor.common.services.shared.validation.model.ValidationMessage;
+import org.jboss.weld.environment.se.Weld;
+import org.jboss.weld.environment.se.WeldContainer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.uberfire.backend.server.util.Paths;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.java.nio.fs.file.SimpleFileSystemProvider;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DRLTextEditorServiceImplCDITest {
+
+    private final SimpleFileSystemProvider fs = new SimpleFileSystemProvider();
+    private WeldContainer weldContainer;
+
+    private List<ValidationMessage> validationMessages;
+    private DRLTextEditorService drlService;
+
+    @Before
+    public void setUp() throws Exception {
+        // disable git and ssh daemons as they are not needed for the tests
+        System.setProperty("org.uberfire.nio.git.daemon.enabled", "false");
+        System.setProperty("org.uberfire.nio.git.ssh.enabled", "false");
+        System.setProperty("org.uberfire.sys.repo.monitor.disabled", "true");
+
+        //Bootstrap WELD container
+        weldContainer = new Weld().initialize();
```

Comment:
For now it's ok like this, but in the future we might want to reuse [this](https://github.com/kiegroup/appformer/blob/master/uberfire-test-utils/src/main/java/org/guvnor/test/CDITestSetup.java) to share this repeated piece of setup (after we simplify it a bit, because [this piece](https://github.com/kiegroup/appformer/blob/master/uberfire-test-utils/src/main/java/org/guvnor/test/CDITestSetup.java#L54-L60) can be simplified to `weldContainer.select(SERVICE_YOU_WANT_TO_INSTANTIATE).get();`).

Reply:
Actually, I remember I sent you link to this, I do not know why I have not used :)